File: data-structures/linear/queue/Queue.go
package main

import (
	"errors"
)

type QueueNode struct {
	data int
	next *QueueNode
}

type Queue struct {
	head *QueueNode
	size int
}

func (q *Queue) Enqueue(data int) {
	newNode := &QueueNode{
		data: data,
	}
	if q.head == nil {
		q.head = newNode
		q.size++
		return
	}

	curr := q.head
	for curr.next != nil {
		curr = curr.next
	}
	curr.next = newNode
	q.size++
}

func (q *Queue) Dequeue() (int, error) {
	if q.head == nil {
		return 0, errors.New("queue is empty")
	}
	val := q.head.data
	q.head = q.head.next
	q.size--
	return val, nil
}

func (q *Queue) Front() (int, error) {
	if q.head == nil {
		return 0, errors.New("queue is empty")
	}
	return q.head.data, nil
}

func (q *Queue) Size() int {
	return q.size
}

func (q *Queue) IsEmpty() bool {
	return q.size == 0
}

File: data-structures/linear/queue/Queue.java
public class Node {
  int data;
  Node next;

  public Node(int data) {
    this.data = data;
    this.next = null;
  }
}

public class Queue {
  private Node head;
  private int size;

  public Queue() {
    head = null;
    size = 0;
  }

  public void enqueue(int data) {
    Node newNode = new Node(data);

    if (head == null) {
      head = newNode;
      size++;
      return;
    }

    Node curr = head;

    while (curr.next != null) {
      curr = curr.next;
    }

    curr.next = newNode;
    size++;
  }

  public void dequeue() {
    if (head == null) {
      System.out.println("Queue is empty");
    }

    head = head.next;
    size--;
  }

  public int front() {
    if (head == null) {
      System.out.println("Queue is empty");
    }

    return head.data;
  }

  public boolean isEmpty() {
    return size == 0;
  }

  public int size() {
    return size;
  }
}
